### PR TITLE
Write readme file for test-frame with usage and examples

### LINKED ISSUE #3: Write a proper readme file

### PR DESCRIPTION
**Pull request by kornys**
close #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# TEST-FRAME
+Library for easy testing of kubernetes deployments and operators using fabric8 api.
```
Comment (im-konge) [inline README.md:2]:
```suggestion
Library for easy testing of Kubernetes deployments and operators using Fabric8 API.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+[![Build](https://github.com/skodjob/test-frame/actions/workflows/build.yaml/badge.svg?branch=main)](https://github.com/skodjob/test-frame/actions/workflows/build.yaml)
+[![Publish-snapshot](https://github.com/skodjob/test-frame/actions/workflows/publish-snapshot.yaml/badge.svg?branch=main)](https://github.com/skodjob/test-frame/actions/workflows/publish-snapshot.yaml)
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+## Provided functionality
+### Kubernetes resource manager
+[KubeResoruceManager](test-frame-common/src/main/java/io/skodjob/testframe/resources/KubeResourceManager.java) provides management of resources created during test phases.
+Every kubernetes resource created by `KubeResoruceManager` is automatically deleted in case of test end or test failed.
```
Comment (im-konge) [inline README.md:11]:
```suggestion
Every Kubernetes resource created by `KubeResoruceManager` is automatically deleted after at the end of the test - no matter if the test passed or failed.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+So kubernetes environment is clean before and after every test run and user do not need to handle it.
```
Comment (im-konge) [inline README.md:12]:
```suggestion
So the Kubernetes environment is clean before and after every test run and user do not need to handle it.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+Working with kubernetes resources using `KubeResoruceManager` also provides proper waiting for resource readiness.
```
Comment (im-konge) [inline README.md:13]:
```suggestion
Working with Kubernetes resources using `KubeResoruceManager` also provides proper wait for resource readiness.
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+### Fabric8 kubernetes client and cmd client
```
Comment (im-konge) [inline README.md:15]:
```suggestion
### Fabric8 Kubernetes client and CMD client
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+Instance of `KubeResoruceManager` contains accessible fabric8 kubernetes client and kubernetes cmd client.
```
Comment (im-konge) [inline README.md:16]:
```suggestion
Instance of `KubeResoruceManager` contains accessible Fabric8 Kubernetes client and Kubernetes CMD client.
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+These clients are initialized and connected to the test cluster based on configuration provided by env file, env variables or kubeconfig.
+
+### Test visual separation
+For better reading test logs library provides ASCII vial separation of tests and test classes.
+
+### Utils
+`TestFrame` contains also tweaks and utils for better working with kubernetes cluster.
+
+
+
```
Comment (im-konge) [inline README.md:26]: extra lines

File: README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+### Fabric8 kubernetes client and cmd client
+Instance of `KubeResoruceManager` contains accessible fabric8 kubernetes client and kubernetes cmd client.
+These clients are initialized and connected to the test cluster based on configuration provided by env file, env variables or kubeconfig.
+
+### Test visual separation
+For better reading test logs library provides ASCII vial separation of tests and test classes.
+
+### Utils
+`TestFrame` contains also tweaks and utils for better working with kubernetes cluster.
```
Comment (im-konge) [inline README.md:23]:
```suggestion
`TestFrame` contains tweaks and utils for better working with Kubernetes cluster.
```

File: README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+
+
+## Usage
+1. Include dependency to your maven test project
```
Comment (im-konge) [inline README.md:28]:
```suggestion
1. Include dependency to your Maven test project
```

File: README.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+```xml
+<dependency>
+    <groupId>io.skodjob</groupId>
+    <artifactId>test-frame-common</artifactId>
+</dependency>
+```
+2. Use annotations for working with `KubeResoruceManager` or other provided functionality
```
Comment (im-konge) [inline README.md:35]:
```suggestion
2. Use annotations for working with `KubeResourceManager` or other provided functionality
```

Comment (im-konge) [reply README.md:35]: `KubeResourceManager` applies to all other places as well.

File: README.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+                new NamespaceBuilder().withNewMetadata().withName("test").endMetadata().build());
+        assertNotNull(KubeResourceManager.getKubeCmdClient().get("namespace", "test"));
+    }
+}
+//...
+```
+
+## Examples
+Examples are stored in [test-frame-test](test-frame-test/src/test/java/io/skodjob/testframe/test/integration) module.
+
```
Comment (im-konge) [inline README.md:62]: extra line

File: README.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+## Config environment variables
+* **ENV_FILE** - path to yaml file with env variables values
```
Comment (im-konge) [inline README.md:65]:
```suggestion
* **ENV_FILE** - path to YAML file with env variables values
```

File: README.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+* **KUBE_USERNAME** - user of the cluster
+* **KUBE_PASSWORD** - password of kube user
```
Comment (im-konge) [inline README.md:67]:
```suggestion
* **KUBE_PASSWORD** - password of Kube user
```

File: README.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+* **KUBE_TOKEN** - token of kube access (use instead of username/password)
```
Comment (im-konge) [inline README.md:68]:
```suggestion
* **KUBE_TOKEN** - token of Kube access (use instead of username/password)
```

File: README.md [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+* **KUBE_URL** - url of the cluster (api url)
```
Comment (im-konge) [inline README.md:69]:
```suggestion
* **KUBE_URL** - URL of the cluster (API URL)
```

File: README.md (hunk 4 above)
Comment (novotnyJiri) [reply README.md]:
```suggestion
Working with Kubernetes resources using `KubeResoruceManager` also provides proper wait mechanism for resource readiness.
```

File: README.md (hunk 2 above)
Comment (obabec) [inline README.md]:
```suggestion
Every Kubernetes resource created by `KubeResoruceManager` is automatically deleted at the end of the test, whether the test passed or failed.
```

File: README.md (hunk 3 above)
Comment (obabec) [reply README.md]:
```suggestion
So Kubernetes environment is clean before and after every test run, and the user does not need to handle it.
```

File: README.md [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+## Provided functionality
+### Kubernetes resource manager
+[KubeResoruceManager](test-frame-common/src/main/java/io/skodjob/testframe/resources/KubeResourceManager.java) provides management of resources created during test phases.
+Every kubernetes resource created by `KubeResoruceManager` is automatically deleted in case of test end or test failed.
+So kubernetes environment is clean before and after every test run and user do not need to handle it.
+Working with kubernetes resources using `KubeResoruceManager` also provides proper waiting for resource readiness.
+
+### Fabric8 kubernetes client and cmd client
+Instance of `KubeResoruceManager` contains accessible fabric8 kubernetes client and kubernetes cmd client.
+These clients are initialized and connected to the test cluster based on configuration provided by env file, env variables or kubeconfig.
```
Comment (obabec) [inline README.md:17]:
```suggestion
These clients are initialized and connected to the test cluster based on the configuration provided by the env file, env variables, or kubeconfig.
```

File: README.md [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+
+### Test visual separation
+For better reading test logs library provides ASCII vial separation of tests and test classes.
```
Comment (obabec) [inline README.md:20]:
```suggestion
For better readability of the test logs, the library provides ASCII vial separation of tests and test classes.
```

Comment (novotnyJiri) [inline README.md:20]:
```suggestion
For better clarity regarding the test logs, `TestFrame` library provides ASCII vial separation of tests and test classes.
```

File: README.md (hunk 8 above)
Comment (novotnyJiri) [reply README.md:23]:
```suggestion
`TestFrame` also contains tweaks and utilities for better kubernetes cluster managment.
```

**Review by im-konge — APPROVED** [review]
Thanks a lot :) good job

**Review by novotnyJiri — APPROVED** [review]

**Review by obabec — APPROVED** [review]

**Pull request merged**